appserver/payara-appserver-modules/microprofile/telemetry/src/main/java/fish/payara/microprofile/telemetry/tracing/OpenTracingHelper.java

```diff
@@ -102,16 +102,10 @@ public String determineOperationName(final ContainerRequestContext request,
             return operationName;
         }
         if (withSpanAnnotation != null) {
-            var operationName = OpenTracingCdiUtils.getConfigOverrideValue(
-                    WithSpan.class, "value", resourceInfo, String.class).orElse(withSpanAnnotation.value());
-
-            // By default, the span name will be <className>.<methodName>,
-            // unless a name is provided as an argument to the annotation.
-            if (operationName.equals("")) {
-                return resourceInfo.getResourceClass().getCanonicalName() + "."
-                        + resourceInfo.getResourceMethod().getName();
-            }
-            return operationName;
+            // replace blank value with <className>.<methodName>,
+            // unless a value is provided as an argument to the annotation.
+            return withSpanAnnotation.value().isBlank() ? resourceInfo.getResourceClass().getCanonicalName() + "."
```

Comment:
Simple `getName` probably serves better than `canonicalName`, as not every class has one.

Many times we need to find first non-synthetic parent of the class, so the name doesn't contain suffix `$$WeldProxy`, but this is likely not the case. It only tends to happen once you have actual reference to object obtained from BeanManager and try to figure out its class.